addons/sourcemod/scripting/include/shavit/sql-create-tables-and-migrations.sp

```diff
@@ -540,6 +554,65 @@ public void Trans_DeprecateExactTimeIntFailed(Database db, ArrayStack stack, int
 	LogError("Timer (core) error! ExactTimeInt migration failed. %d %d Reason: %s", numQueries, failIndex, error);
 }
 
+void ApplyMigration_FixSQLiteMapzonesROWID()
+{
+	if (gI_Driver != Driver_sqlite)
+	{
+		InsertMigration(Migration_FixSQLiteMapzonesROWID);
+		return;
+	}
+
+	char sQuery[256];
+	FormatEx(sQuery, sizeof(sQuery), "SELECT EXISTS(SELECT 1 FROM `%smapzones` WHERE id IS NULL);", gS_SQLPrefix);
```

Comment:
Do you think it would be fine to replace this with `SELECT 1 FROM mapzones WHERE id IS NULL LIMIT 1;`?

Reply:
Like the original query, this doesn't handle the scenario I mentioned above (empty mapzones table with the old table creation statement). So ideally you'd need to directly test id's type to definitively know if the migration is required or not.

You could theoretically replace the query using a [PRAGMA function](https://www.sqlite.org/pragma.html), but it is listed as experimental so might break in the future:
```suggestion
	FormatEx(sQuery, sizeof(sQuery), "SELECT 1 FROM pragma_table_info('%smapzones') WHERE name = 'id' AND type = 'INT AUTO_INCREMENT' LIMIT 1;", gS_SQLPrefix);
```
This is the only way of accessing PRAGMA results with a SELECT statement (i.e. `SELECT * FROM PRAGMA table_info(mapzones) WHERE ...;` doesn't work). Then again, the docs do say that "specific pragma statements may be removed and others added in future releases of SQLite. There is no guarantee of backwards compatibility" so probably avoiding PRAGMA statements and functions is the best move long term.

Which means, as far as I can tell, the best way of testing id's type is to use `SELECT sql FROM sqlite_master` + `StrContains` as I did in #1176. Let me know if you agree with that approach and I'll commit it.

Reply:
OK, done. Should be good now.